src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessible.java

```diff
@@ -182,17 +182,23 @@ public void propertyChange(PropertyChangeEvent e) {
 
                     // Do send check box state changes to native side
                     if (thisRole == AccessibleRole.CHECK_BOX) {
-                        valueChanged(ptr);
+                        if (newValue != null && !newValue.equals(oldValue)) {
```

Comment:
Is it possible to have "newValue == null" and the "oldValue != null"? Do we need to update the native in this case?

Reply:
No. Null value is not a correct value to be set for these types of control so if null is passed then it is not a real value update event.

Reply:
[`Objects.equals`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Objects.html#equals(java.lang.Object,java.lang.Object)) can be used to avoid null-check, it handles `null` values gracefully.